src/study/system-model/include/antares/study/system-model/portType.h

```diff
@@ -40,11 +40,7 @@ class PortType
         id_(id),
         fields_(std::move(fields))
     {
-        if (areaConnectionFieldId.empty())
-        {
-            areaConnectionFieldId_ = std::nullopt;
-        }
-        else
+        if (!areaConnectionFieldId.empty())
```

Comment:
In the end, do we need the `if (!areaConnectionFieldId.empty())` ?
I mean, would it make sense to have : 
```c++
PortType(const std::string& id,
         std::vector<PortField>&& fields,
         const std::string& areaConnectionFieldId = ""):
    id_(id),
    fields_(std::move(fields))
{
    if (!std::ranges::any_of(fields_,
                             [areaConnectionFieldId](const auto& field)
                             { return field.Id() == areaConnectionFieldId; }))
    {
         throw std::invalid_argument(
           "Field \"" + areaConnectionFieldId
           + "\" selected for area connections was not defined in PortType \"" + id_
           + "\".");
    }
    areaConnectionFieldId_ = std::optional(areaConnectionFieldId);
}
```
?

Reply:
you're right, a field ID can't be empty

Reply:
what you suggest would throw the exception if areaConnectionFieldId="" (which is the default case, when the port type can't be connected to an area)

Reply:
but I can merge the if statements at least

Reply:
ok, i misunderstood.